10/buster-slim/Dockerfile

```diff
@@ -47,7 +47,10 @@ RUN buildDeps='xz-utils' \
 
 ENV YARN_VERSION 1.21.1
 
-RUN set -ex \
+RUN buildDeps='ca-certificates curl wget gnupg dirmngr' \
+  && set -ex \
+  && apt-get update && apt-get install -y $buildDeps --no-install-recommends \
```

Comment:
Does it need to install again for the second RUN?

Reply:
Or should the apt-get purge just be moved down

Reply:
It can't be moved down because of how layers work. It should be reinstalled again in the yarn step.

Reply:
OK, I thought if the apt-get purge was removed from the previous RUN block, the packages would be available when running this layer too

Reply:
the layers will be larger though. Ideally we'd have one large `RUN` which install both node and yarn in the same layer, but I didn't want to make such a big change.

Reply:
@nschonni it package would be available but wouldn't be removable because they'd be "baked in" the previous layer.

Reply:
Ah, thanks for clearing that up

Reply:
It's the downside of the power of docker layers

Reply:
In many repositories we've stopped using an explicit list of "things to install then remove later" exactly because of this problem of dependencies that we didn't install getting removed (or hand-maintaining a list of library dependencies after we've compiled something using them), instead opting to get "clever" with `apt-mark` to let APT itself do the heavy lifting (which then will only remove things we haven't explicitly said need to stay):

```dockerfile
RUN set -eux; \
	savedAptMark="$(apt-mark showmanual)"; \
	apt-get update; \
	apt-get install -y --no-install-recommends ca-certificates curl dirmngr gnupg wget; \
	rm -rf /var/lib/apt/lists/*; \
	\
# ... do stuff here ...
	\
	apt-mark auto '.*' > /dev/null; \
	apt-mark manual $savedAptMark > /dev/null; \
	apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false
```

See https://github.com/docker-library/php/blob/ae3f372b446b6457326504d9b9984ce708f20c3d/Dockerfile-debian.template#L65-L97 for one example of this pattern being used, and https://github.com/docker-library/php/blob/ae3f372b446b6457326504d9b9984ce708f20c3d/Dockerfile-debian.template#L196-L207 for a more complicated example of keeping appropriate `lib*` packages automatically.

Reply:
Ah, nice! I've never even heard of `apt-mark` 😅 I'll change to use that, seems ideal

Reply:
Pushed that now. `apt-mark showmanual` was empty, so I dropped that part